android/src/main/java/com/reactnative/googlefit/StepHistory.java

```diff
@@ -172,16 +172,35 @@ public void aggregateDataByDate(long startTime, long endTime, int bucketInterval
 
             Log.i(TAG, "  + Device    : " + device);
             if (device != null) {
+                source.putString("deviceUid", device.getUid());
                 source.putString("deviceManufacturer", device.getManufacturer());
                 source.putString("deviceModel", device.getModel());
                 switch(device.getType()) {
+                    case Device.TYPE_PHONE:  
+                        source.putString("deviceType", "phone");
+                    break;
+                    case Device.TYPE_WATCH: 
+                        source.putString("deviceType", "watch");
+                    break;
+                    case Device.TYPE_TABLET: 
+                        source.putString("deviceType", "tablet");
+                    break;
                     case Device.TYPE_CHEST_STRAP:
-                        source.putString("deviceType", "chestStrap"); break;
+                        source.putString("deviceType", "chest-strap");
```

Comment:
I changed the format to "kebab case" since I think that it's easier to process than separating the words with an uppercase letter.

Reply:
I suggest just use the old format to keep the consistency between both js side and native module. So people in the future wouldn't get confuse if they want to update the code.

Reply:
> People always had their own judge on naming. Some might prefer camel case. So they wouldn't get confuse if they want to update the code. Otherwise they may assume this is particular specification from Google side.

The only reason why I changed it was that using camel case for values feels really strange, and the Web/React libraries almost always use kebab-case for values.

I can change it back to camel case, but I just want to point out that the naming rules in my opinion should follow the naming convention which is common React and React Native, not a personal preference of someone.

Reply:
I do understand your intent and indeed it looks better in terms of readability.
React does has certain rules for naming but mainly for file system and component as you claimed. 
But my take is that I don't know the string value will be directly used for display, or still be parsed by devs who used this library, which will have some fundamental difference for me when I decided the name. By the way, this part of code wasn't written by me.

So no matter how the string naming end up, 
in user(dev) side, based on my past experience, it will always become something like
```javascript
switch(deviceType) {
  case(string name for watch): return <Icon class="watch"/> 
}
```
 so it's technically a string but used as a variable. But again, it's string like you say,

I personally don't have any best practice, it's always up to the team and the project to figure out the middle/comfortable point to work with without losing the integrity. For this project, things can be weird I guess it's not just about react native, we had to find a sweet spot in terms of language (java & javascript).

Reply:
I'm not saying that using camel case in this case is wrong, but I'm just trying to advocate the common practices in the Javascript world.

I've been writing Javascript for 20 years, and I can assure you that camel case for values not a common practice for the Web/Javascript APIs (though there are obviously exceptions).

Here's how the naming is *usually* done for Web APIs:

- Class: pascal case
- Event: lowercase
- Function/Method: camel case
- Property: camel case
- String value: kebab case or lowercase
 
You can just open MDN and search for any Web API, and see how they're defined (should follow the same logic). 

For example the Fetch API, which has the option properties in camel case, and the values in kebab case: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options

Reply:
Actually, fetch Api example is quite convincing for me. 
Do you might just reserve the kebab-case back. 
I was also thinking about formatting the codebase at some point to keep certain consistency,  as I get rid of the deprecated code.